src/components/AcceptProposalForm.tsx

```diff
@@ -12,12 +12,16 @@ interface AcceptProposalFormProps {
   proposalName: string
   proposalId: string
   supervisorEmail: string
+  feedbackGiven: boolean
+  setFeedbackGiven: (value: boolean) => void
 }
 
 export default function AcceptProposalForm({
   proposalName,
   proposalId,
   supervisorEmail,
+  feedbackGiven,
+  setFeedbackGiven,
 }: AcceptProposalFormProps) {
```

Comment:
place the useSessionStorage hook inside the forms to have access to the proposalId. when using proposalId as the key, all forms would read from the same storage key, and you could automatically hide the form in all components when a feedback has been given in any of them